core-rust/core-api-server/src/core_api/handlers/stream_proofs.rs

```diff
@@ -0,0 +1,211 @@
+
+    let page_size = extract_valid_size(
+        request.max_page_size,
+        SizeRange {
+            min: 1,
+            default: 20,
+            max: 100,
+        },
+    )
+    .map_err(|err| err.into_response_error("max_page_size"))?;
+
+    let continue_from_state_version =
+        extract_continuation_token::<StateVersion>(request.continuation_token)
+            .map_err(|err| err.into_response_error("continuation_token"))?;
+
+    let database = state.state_manager.database.read_current();
+
+    use models::StreamProofsFilter::*;
+    let mut proofs_iter = match filter {
+        StreamProofsFilterAll { from_state_version } => iterate_all_proofs(
+            &database,
+            continue_from_state_version,
+            extract_from_state_version(&database, from_state_version)?,
+        ),
+        StreamProofsFilterNewEpochs { from_epoch } => iterate_end_of_epoch_proofs(
+            &database,
+            continue_from_state_version,
+            extract_from_epoch(&mapping_context, &database, from_epoch)?,
+        ),
+        StreamProofsFilterProtocolUpdateInitializations { from_state_version } => {
+            iterate_protocol_update_initialization_proofs(
+                &database,
+                continue_from_state_version,
+                extract_from_state_version(&database, from_state_version)?,
+            )
+        }
+        StreamProofsFilterProtocolUpdateExecution {
+            from_state_version,
+            protocol_version,
+        } => iterate_protocol_update_execution_proofs(
+            &database,
+            continue_from_state_version,
+            extract_from_state_version(&database, from_state_version)?,
+            protocol_version,
+        ),
+    }?;
+
+    let (page, continuation_token) = to_api_page(
+        &mut proofs_iter,
+        page_size,
+        |proof| handlers::to_api_ledger_proof(&mapping_context, proof),
+        |proof| proof.ledger_header.state_version,
+    )?;
+
+    Ok(Json(models::StreamProofsResponse {
+        page,
+        continuation_token,
+    }))
+}
+
+fn iterate_all_proofs<'a>(
+    database: &'a StateManagerDatabase,
+    continue_from_state_version: Option<StateVersion>,
+    from_state_version: StateVersion,
+) -> Result<
+    Box<dyn Iterator<Item = LedgerProof> + 'a>,
+    ResponseError<models::StreamProofsErrorDetails>,
+> {
+    let start = optional_max(from_state_version, continue_from_state_version);
+
+    Ok(database.get_proof_iter(start))
+}
+
+fn iterate_end_of_epoch_proofs<'a>(
+    database: &'a StateManagerDatabase,
+    continue_from_state_version: Option<StateVersion>,
+    from_epoch: Epoch,
+) -> Result<
+    Box<dyn Iterator<Item = LedgerProof> + 'a>,
+    ResponseError<models::StreamProofsErrorDetails>,
+> {
+    let continuation_next_epoch = match continue_from_state_version {
+        Some(state_version) => match database.get_proof_iter(state_version).next() {
```

Comment:
looks like we could just use next epoch as a continuation token for this filter

Reply:
but that requires a separate `to_api_page` call for different filter types above, so prolly not worth it :)

Reply:
Yeah I considered it! Maybe more natural :D. (see thread: https://github.com/radixdlt/babylon-node/pull/819#discussion_r1465796089 )

But I don't think worth changing for now given we're trying to get in for anemone.